src/ir/selector_access_path.h

```diff
@@ -0,0 +1,85 @@
+#ifndef SRC_IR_SELECTOR_ACCESS_PATH_H_
+#define SRC_IR_SELECTOR_ACCESS_PATH_H_
+
+#include "selector.h"
+
+#include "absl/hash/hash.h"
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+
+#include <vector>
+
+namespace raksha::ir {
+
+// Represents an SelectorAccessPath descending through the members of types in a
+// Manifest. This is eventually used for printing edges to the resultant Datalog
+// file.
+//
+// We use this class instead of a bare vector for the following reasons:
+//
+// 1. Encapsulating the reversed vector prevents us from having to reason
+// about this all the time and potentially be confused by it.
+//
+// 2. Adding our own containing struct allows us to force copies to be
+// explicitly requested by making the copy constructor private, but
+// accessible with a public "copy" function. Because we are returning this
+// object owning variable-length memory in a bottom-up fashion, copies could
+// cause us to have quadratic performance.
+//
+// 3. Providing our own interface allows us to make this class immutable
+// except for move constructors. We define no "add", "insert", etc. methods;
+// "mutation" is done only by constructing a new object from a previous
+// object. That prevents many forms of state-manipulation bugs.
+//
+// 4. The name SelectorAccessPath is a bit more self-documenting.
+class SelectorAccessPath {
+ public:
+  // Create a leaf SelectorAccessPath from a single leaf selector.
+  explicit SelectorAccessPath(const Selector leaf) : reverse_selectors_{leaf} {}
```

Comment:
Should this be `const Selector&& leaf`?

Reply:
I'll make the copy constructors private on `Selector`s just to make sure I'm doing this right. In C++17, copy elision is much more powerful and you can use value types in many cases where you had to use `&&` before. But I might need to make the parameter not `const` and use `std::move` in that use of `leaf`.

Reply:
Addressed in #68.